middle_end/flambda2/reaper/traverse_acc.ml

```diff
@@ -0,0 +1,239 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*           Nathanaëlle Courant, Pierre Chambart, OCamlPro               *)
+(*                                                                        *)
+(*   Copyright 2024 OCamlPro SAS                                          *)
+(*   Copyright 2024 Jane Street Group LLC                                 *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+module Graph = Global_flow_graph
+
+type code_dep =
+  { arity : [`Complex] Flambda_arity.t;
+    params : Variable.t list;
+    my_closure : Variable.t;
+    return : Variable.t list; (* Dummy variable representing return value *)
+    exn : Variable.t (* Dummy variable representing exn return value *)
+  }
+
+type apply_dep =
+  { apply_in_func : Code_id.t option;
+    apply_code_id : Code_id.t;
+    apply_params : Simple.t list;
+    apply_closure : Simple.t option;
+    apply_return : Variable.t list option;
+    apply_exn : Variable.t
+  }
+
+type t =
+  { mutable code : code_dep Code_id.Map.t;
+    mutable apply_deps : apply_dep list;
+    mutable set_of_closures_dep : (Name.t * Code_id.t) list;
```

Comment:
Maybe put pair in a record; `let_bound_name_of_the_closure` and `closure_code_id`.